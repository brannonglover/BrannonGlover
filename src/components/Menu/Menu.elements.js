import styled from 'styled-components/macro';

export const MenuWrapper = styled.div`
  width: 50%;
  position: fixed;
  top: 4em;
  right: -160px;
  background-color: #414047;
  transition: right .3s ease;

  &.showMenu {
    right: 0;
    transition: right .3s ease;
  }

  ul {
    list-style-type: none;

    li {
      color: #fff;
      padding: .5em 0;
    }
  }
`;