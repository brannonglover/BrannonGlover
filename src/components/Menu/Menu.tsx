import React from "react";
import { MenuWrapper } from "./Menu.elements";
import data from "../../assets/info.json";

interface IMenuProps {
  showMenu?: boolean;
  menuToggle?: () => void;
}

const Menu: React.FC<IMenuProps> = ({ showMenu, menuToggle }) => {
  return (
    <MenuWrapper onClick={menuToggle} className={showMenu ? "showMenu" : ""}>
      <ul>
        {data.links.map((link) => (
          <li key={link.id}>
            <a href={link.url} rel="noreferrer" target="_blank">
              {link.name}
            </a>
          </li>
        ))}
      </ul>
    </MenuWrapper>
  );
};

export default Menu;
