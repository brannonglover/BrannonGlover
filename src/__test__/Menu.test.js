import React from 'react';
import { render } from '@testing-library/react';
import Menu from '../components/Menu/Menu';

test('Render Menu component', () => {
  const { asFragment } = render(<Menu />);

  expect(asFragment()).toMatchSnapshot();
});