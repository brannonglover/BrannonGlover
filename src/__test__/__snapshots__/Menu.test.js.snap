// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Menu component 1`] = `
<DocumentFragment>
  <div
    class="Menuelements__MenuWrapper-k5kaq1-0 httLqd"
  >
    <ul>
      <li>
        <a
          href="https://github.com/brannonglover?tab=repositories"
          rel="noreferrer"
          target="_blank"
        >
          GitHub
        </a>
      </li>
      <li>
        <a
          href="https://twitter.com/BrannonGlover"
          rel="noreferrer"
          target="_blank"
        >
          Twitter
        </a>
      </li>
      <li>
        <a
          href="https://medium.com/@brannonglover"
          rel="noreferrer"
          target="_blank"
        >
          Medium
        </a>
      </li>
      <li>
        <a
          href="https://www.linkedin.com/in/brannon-glover-31a22212/"
          rel="noreferrer"
          target="_blank"
        >
          LinkedIn
        </a>
      </li>
    </ul>
  </div>
</DocumentFragment>
`;
